src/lucky_record/associations.cr

```diff
@@ -9,6 +14,9 @@ module LuckyRecord::Associations
 
   macro belongs_to(type_declaration)
     {% assoc_name = type_declaration.var }
+    {% LuckyInflector::Inflector.pluralize assoc_name %} # Can't get this to work...
+
+    association {{ assoc_name }}, :id
```

Comment:
This is what really frustrated me. `LuckyInflector::Inflector` is a `TypeNode` so I can't call it's methods, only list them through `class.methods`. But I need to somehow pluralize the `belongs_to` association for the generated methods like `join_posts` and for the join statement itself.

Any thoughts?

Reply:
Ah yeah this is a bit tricky. You can’t run any runtime code in a macro call. There some ways to do it, but i think there is a better option in this case. 

In the model we set @@table_name. We can also set a TABLE_NAME constant. This is because constants can be accessed in macros by calling .constant(:THE_CONSTANT_NAME)

So, if we set the TABLE_NAME constant we could get the associations table name in this macro like this: `type_declaration.type.constant(:TABLE_NAME)`

Does that make sense?

Reply:
That's perfect. I dug through the `TypeNode` documentation and it never occurred to me to use a constant. Thanks for this, I'm sure I'll find it useful in the future.